### PR TITLE
Update base FileReader to create a generator.

### PR DESCRIPTION
**Pull request by slacrherbst**
This will break existing uses, but I don't think anyone is using this yet.]

Example usage:

```python
        with FileReader(fileList,configChan=1) as fr:

            for header,data in fr.records():

                 # Process data as uint8 numpy array

                 val = fr.configValue('root.blah.blah')  # Get current config state

```

### CODE REVIEW

File: python/pyrogue/utilities/fileio/_FileReader.py [hunk 1]
```diff
@@ -16,74 +16,116 @@
...
-class FileReader(object):
+RogueHeaderSize  = 8
+RogueHeaderPack  = 'IHBB'
 
-    def __init__(self, filename, configChan=None):
-        self._filename   = filename
-        self._configChan = configChan
+# Default header as a named tuple
+RogueHeader = namedtuple( 'RogueHeader',
+                         [ 'size'                ,  # 4 Bytes, uint32, I
```
Comment (ruck314) [inline python/pyrogue/utilities/fileio/_FileReader.py]: Does size of 32-bit (4 bytes) means there a 4GB limit for the FileReader/FileWriter?  How does the FileReader/FileWriter handle the > 4GB case?

Comment (slacrherbst) [reply python/pyrogue/utilities/fileio/_FileReader.py]: The 32-bit size field is each record. I hope we don't have records that large.

Comment (ruck314) [reply python/pyrogue/utilities/fileio/_FileReader.py]: Does the FileWriter close the file and stop when the 4GB limit is reached?

**Comment by slacrherbst** [comment]
The FileWriter can break up the files. But remember the size is each single Frame object written to the file. Do we expect to ever have 4G frames? This is independent of the total file size.

**Pull request merged**